### PR TITLE
[image_picker] Removes use of PHAsset when using PHPickerViewController

### PR DESCRIPTION
**Pull request by WenHaozhan**
Fixes issue https://github.com/flutter/flutter/issues/90373

Currently on IOS 14+, images are picked using `PHPickerViewController` which does not need photo permissions and also gets the full image metadata regardless of `requestFullMetadata`. However, it currently retrieves the metadata using `PHAsset` which does require permission and causes the gallery opening twice issue. Another issue is that an error is thrown when permission is denied even if none are required. 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version
[following repository CHANGELOG style]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style
[exempt from CHANGELOG changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog
[test-exempt]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests

### CODE REVIEW

**Comment by WenHaozhan** [comment]
Any update on this?

**Comment by loic-sharma** [comment]
@WenHaozhan Hello, this is on my queue to review. I'm not familiar with the image picker plugin yet and need to ramp up first before I'm able to review. I apologize for the delay.

File: packages/image_picker/image_picker_ios/example/ios/RunnerTests/ImagePickerPluginTests.m [hunk 1]
```diff
@@ -529,11 +529,19 @@ - (void)testPickImageAuthorizationDenied API_AVAILABLE(ios(14)) {
                       quality:nil
                  fullMetadata:YES
                    completion:^(NSString *result, FlutterError *error) {
+                     // Does not error due to access denied
```
Comment (loic-sharma) [inline packages/image_picker/image_picker_ios/example/ios/RunnerTests/ImagePickerPluginTests.m:532]: Should we remove this test entirely as this is no longer a concern?

File: packages/image_picker/image_picker_ios/ios/image_picker_ios/Sources/image_picker_ios/FLTImagePickerPlugin.m [hunk 2]
```diff
@@ -113,11 +113,7 @@ - (void)launchPHPickerWithContext:(nonnull FLTImagePickerMethodCallContext *)con
   pickerViewController.presentationController.delegate = self;
   self.callContext = context;
 
-  if (context.requestFullMetadata) {
-    [self checkPhotoAuthorizationWithPHPicker:pickerViewController];
```
Comment (loic-sharma) [inline packages/image_picker/image_picker_ios/ios/image_picker_ios/Sources/image_picker_ios/FLTImagePickerPlugin.m:117]: Does anything use `checkPhotoAuthorizationWithPHPicker` now? Can we remove it?

File: packages/image_picker/image_picker_ios/ios/image_picker_ios/Sources/image_picker_ios/FLTPHPickerSaveImageToPathOperation.m [hunk 3]
```diff
@@ -128,64 +128,20 @@ - (void)start {
 - (void)processImage:(NSData *)pickerImageData API_AVAILABLE(ios(14)) {
   UIImage *localImage = [[UIImage alloc] initWithData:pickerImageData];
 
-  PHAsset *originalAsset;
-  // Only if requested, fetch the full "PHAsset" metadata, which requires  "Photo Library Usage"
-  // permissions.
-  if (self.requestFullMetadata) {
-    originalAsset = [FLTImagePickerPhotoAssetUtil getAssetFromPHPickerResult:self.result];
```
Comment (loic-sharma) [inline packages/image_picker/image_picker_ios/ios/image_picker_ios/Sources/image_picker_ios/FLTPHPickerSaveImageToPathOperation.m:135]: Does anything use `getAssetFromPHPickerResult` now? Can we remove it?

File: packages/image_picker/image_picker_ios/CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,7 @@
+## 0.8.12+2
+
+* Removes use of PHAsset when using PHPickerViewController in image_picker to remove needing permissions.
```
Comment (loic-sharma) [inline packages/image_picker/image_picker_ios/CHANGELOG.md:3]: What do you think of this?

```suggestion
* Removes the need for user permissions to pick an image on iOS 14+.
```

File: packages/image_picker/image_picker_ios/example/ios/RunnerTests/ImagePickerPluginTests.m (hunk 1 above)
Comment (WenHaozhan) [reply packages/image_picker/image_picker_ios/example/ios/RunnerTests/ImagePickerPluginTests.m]: I removed this test.

File: packages/image_picker/image_picker_ios/ios/image_picker_ios/Sources/image_picker_ios/FLTImagePickerPlugin.m (hunk 2 above)
Comment (WenHaozhan) [reply packages/image_picker/image_picker_ios/ios/image_picker_ios/Sources/image_picker_ios/FLTImagePickerPlugin.m:117]: Don't think anything uses this, removed.

File: packages/image_picker/image_picker_ios/ios/image_picker_ios/Sources/image_picker_ios/FLTPHPickerSaveImageToPathOperation.m (hunk 3 above)
Comment (WenHaozhan) [reply packages/image_picker/image_picker_ios/ios/image_picker_ios/Sources/image_picker_ios/FLTPHPickerSaveImageToPathOperation.m:135]: Removed function and a test that uses it.

**Comment by domesticmouse** [comment]
I'm not required now that `flutter_svg` and `flutter_markdown` have been removed

**Comment by WenHaozhan** [comment]
Hey, sorry about this, I messed up a rebase. I will create a new pull request.

**Pull request closed without merging**